docs/design/direct-assign-volume.md

```diff
@@ -0,0 +1,139 @@
+should be mounted within the container(s).
+
+### Specifiying that the volume will be consumed by a VM-friendly runtime
+
+If a persistent volume [(PV)](https://kubernetes.io/docs/concepts/storage/persistent-volumes/) is intended to be consumed by a VM-based runtime, the user can specify this by adding an annotation to their respect persistent volume claim (PVC). Example:
+
+```yaml
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  annotations:
+    skip-hostmount: "true"
+  name: amazing-kata-volume
+spec:
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 4Gi
+```
+
+Based on this annotation, the CSI node driver would identify that the volume should not be mounted, and special handling is needed.
+
+### Communicating block device details to the runtime
+
+The runtime will need to identify that a mount is backed by a block device and that it needs to handle the direct-assignment use case.
+
+When a volume is presented for direct-assignment, the CSI driver will create a particular file, `csiPlugin.json`
+at the root of the volume on the host. The contents of this file will be defined as follows:
+
+```go
+type DiskMountInfo struct {
+        // Device: source device for this volume
+        Device string json:"device,omitempty"
+
+        // VolumeType: type associated with this volume (ie, block)
+        VolumeType string json:"volumeType,omitempty"
+
+        // TargetPath: path which this device should be mounted within the guest
+        TargetPath string json:"targetPath,omitempty"
+
+        // FsType: filesystem that needs to be used to mount the storage inside the VM
+        FsType string json:"fsType,omitempty"
+
+        // Options: additional options that might be needed to mount the storage filesystem
+        Options string json:options,omitempty"
+}
+```
+
+The block device will be formatted, but not mounted on the host, by the CSI node driver. Through `csiPlugin.json`,
+the runtime will identify that the existing mount should not be mounted into the guest by way of the shared filesystem,
+but instead the described device should be attached to the VM, and volume mount information should be provided to the
+agent for mounting the in-guest block device to the TargetPath in the guest for consumption by the container(s) in
+the pod.
+
+## Managing the mount
+
+Normally any node agent, including the CSI node driver, would be able to directly check the status of the mountpath, 
+and look at usage statistics. When the volume mount is not mounted on the host, this is no longer feasible. We need
+to introduce a mechanism for gathering similar information on nodes that are utilizing Kata Containers with
+direct-assigned volumes.
+
+These actions will be achieved by interacting with a host binary, `kata-runtime`, which will in turn interact with
+the applicable shim-kata-v2 instance on the host which will interact with the applicable in-guest kata-agent.
```

Comment:
- I'd replace "shim-kata-v2" with something like "Kata Containers `shimv2` instance".
- s/kata-agent/Kata agent/ (or `kata-agent`, but I think the text name is better here).